tests/components/drop_connect/test_sensor.py

```diff
@@ -0,0 +1,319 @@
+    psi_sensor = hass.states.get(psi_sensor_name)
+    assert psi_sensor
+    assert round(float(psi_sensor.state), 1) == 348.2  # centibars
+
+    capacity_sensor = hass.states.get(capacity_sensor_name)
+    assert capacity_sensor
+    assert round(float(capacity_sensor.state), 1) == 3785.4  # liters
+
+
+async def test_sensors_filter(
+    hass: HomeAssistant, config_entry_filter, mqtt_mock: MqttMockHAClient
+) -> None:
+    """Test DROP sensors for filters."""
+    config_entry_filter.add_to_hass(hass)
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+
+    battery_sensor_name = "sensor.filter_battery"
+    hass.states.async_set(battery_sensor_name, STATE_UNKNOWN)
+    current_flow_sensor_name = "sensor.filter_water_flow_rate"
+    hass.states.async_set(current_flow_sensor_name, STATE_UNKNOWN)
+    psi_sensor_name = "sensor.filter_current_water_pressure"
+    hass.states.async_set(psi_sensor_name, STATE_UNKNOWN)
+
+    async_fire_mqtt_message(hass, TEST_DATA_FILTER_TOPIC, TEST_DATA_FILTER_RESET)
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(hass, TEST_DATA_FILTER_TOPIC, TEST_DATA_FILTER)
+    await hass.async_block_till_done()
+
+    battery_sensor = hass.states.get(battery_sensor_name)
+    assert battery_sensor
+    assert round(float(battery_sensor.state), 1) == 12.0
+
+    current_flow_sensor = hass.states.get(current_flow_sensor_name)
+    assert current_flow_sensor
+    assert round(float(current_flow_sensor.state), 1) == 19.8
+
+    psi_sensor = hass.states.get(psi_sensor_name)
+    assert psi_sensor
+    assert round(float(psi_sensor.state), 1) == 263.4  # centibars
+
+
+async def test_sensors_protection_valve(
+    hass: HomeAssistant, config_entry_protection_valve, mqtt_mock: MqttMockHAClient
+) -> None:
+    """Test DROP sensors for protection valves."""
+    config_entry_protection_valve.add_to_hass(hass)
+    assert await async_setup_component(hass, DOMAIN, {})
+    await hass.async_block_till_done()
+
+    battery_sensor_name = "sensor.protection_valve_battery"
+    hass.states.async_set(battery_sensor_name, STATE_UNKNOWN)
+    current_flow_sensor_name = "sensor.protection_valve_water_flow_rate"
+    hass.states.async_set(current_flow_sensor_name, STATE_UNKNOWN)
+    psi_sensor_name = "sensor.protection_valve_current_water_pressure"
+    hass.states.async_set(psi_sensor_name, STATE_UNKNOWN)
+    temp_sensor_name = "sensor.protection_valve_temperature"
+    hass.states.async_set(temp_sensor_name, STATE_UNKNOWN)
+
+    async_fire_mqtt_message(
+        hass, TEST_DATA_PROTECTION_VALVE_TOPIC, TEST_DATA_PROTECTION_VALVE_RESET
+    )
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(
+        hass, TEST_DATA_PROTECTION_VALVE_TOPIC, TEST_DATA_PROTECTION_VALVE
+    )
+    await hass.async_block_till_done()
+
+    battery_sensor = hass.states.get(battery_sensor_name)
+    assert battery_sensor
+    assert round(float(battery_sensor.state), 1) == 0
```

Comment:
I think you don't have to round and convert it to a float

Reply:
I simplified the asserts for integer values in 3689d81d2e5b2be235f4770f997c918b50aa04c5.  The round(float()) on many of the asserts are there because the native sensor units are converted to internal units on the fly (e.g., psi to centibars or gallons to liters) and I want to avoid having tests like `assert used_today_sensor.state == '881.35298873219472'`.